### PR TITLE
Delete persons endpoint

### PR DESCRIPTION
**Pull request by sher812**
Added Delete Persons Endpoint:

Deletes specified person (using objectID). Also removes persons from related encounters/users. Empty encounters are deleted.

Manually tested endpoints with postman.

Fixed(#55)

People who worked on this issue:
@UoA-sjer215 and @sher812 

NOTE: 

This PR is currently blocked by issue (#41). So will need to wait until that PR is merged before this one can be.

### CODE REVIEW

File: backend/src/services/encounter.service.ts [hunk 1]
```diff
@@ -12,9 +12,21 @@ const updateEncounter = async (objectID: string, encounterDetails: EncounterMode
     return updatedEncounter;
 }
 
+const deleteEncounterPerson = async (personID: string) => {
+  // Service is used for deletePersons endpoint
+
+  await Encounter.updateMany({ }, { $pullAll: {persons: [{ _id: personID}]} });
+  const empty_encounters = await Encounter.find({persons : {$exists:true, $size:0}});
+  await Encounter.deleteMany({persons: {$exists: true, $size: 0}});
+
```
Comment (sdiv877) [inline backend/src/services/encounter.service.ts:21]: Might be nice to have a few comments for this part such as:
* Remove the person with the supplied ID from all existing encounters
* Store references to the encounters that no longer contain persons so they can be removed from Users later
* Delete all the empty encounters

**Review by sdiv877 — APPROVED** [review]
LGTM! 👍

File: backend/src/controllers/encounter.controller.ts [hunk 2]
```diff
@@ -16,11 +16,8 @@ import {getUserByAuthId} from "../services/user.service";
    logger.info("POST /encounter request from frontend");
  
    try {
-     // Grab the data from the req
-     const encounterReq = getEncounterFromReqBody(req.body);
-
      // Pass data to service and attempt to save
-     const createdEncounter = await encounterService.createEncounter(encounterReq);
+     const createdEncounter = await encounterService.createEncounter(req.body);
```
Comment (AdamAWiener) [inline backend/src/controllers/encounter.controller.ts:20]: Are we adding a reference to this `encounter` to the `user` and `person` that owns it? I can't seem to find that.

Comment (sher812) [reply backend/src/controllers/encounter.controller.ts:20]: @AdamAWiener sorry think I messed up from a previously merged issue. I think this createEncounters() was merged from the issue (#94). Currently @schi314 is working on the createEncounters() issue (#41). Should I remove the createEncounters() from my PR?

**Review by jimwang6012 — DISMISSED** [review]
LGTM!

File: backend/src/controllers/encounter.controller.ts (hunk 2 above)
Comment (AdamAWiener) [reply backend/src/controllers/encounter.controller.ts:20]: Yeah maybe add a comment saying this PR is blocked by #41 and wait until that's merged in to get access to the method.

**Comment by sher812** [comment]
Resolved merge conflicts

**Review by jimwang6012 — DISMISSED** [review]
LGTM!

**Comment by sdiv877** [comment]
LGTM! 👍

**Review by sdiv877 — APPROVED** [review]

**Review by sdiv877 — APPROVED** [review]
LGTM! 👍

**Review by butterplup — APPROVED** [review]
LGTM

**Pull request merged**